### PR TITLE
Dev refactor

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: client/src/main/java/model/Ship.java [hunk 1]
```diff
@@ -40,12 +37,12 @@ void addMast(Field mast) {
     return masts.stream().map(Field::positionAsInteger).collect(Collectors.toList());
   }
 
+  Field lastMast() {
+    return masts.get(masts.size() -1);
```
Comment (Ozymadias) [inline client/src/main/java/model/Ship.java]: space before 1 if you like

File: client/src/test/java/model/DataProvider.java [hunk 2]
```diff
@@ -39,4 +65,18 @@
...
 
+  @org.testng.annotations.DataProvider(name = "positions")
+  public static Object[][] positions() {
+
+    int testCount = 10;
+    int boundForRandom = 99;
+    Object[][] positionsIndexes = new Object[testCount][];
+
+    for(int i = 0; i < testCount; i++) {
+      Random random = new Random();
```
Comment (Ozymadias) [inline client/src/test/java/model/DataProvider.java]: I my opinion random values shouldn't be used in test

File: client/src/test/java/model/DummyClickableField.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    this.row = position / boardSize;
+  }
+
+  @Override
+  public void makeUnclickable() {
+  }
+
+  @Override
+  public void makeClickable() {
+  }
```
Comment (Ozymadias) [inline client/src/test/java/model/DummyClickableField.java]: why this methods are empty?

File: client/src/test/java/model/DataProvider.java [hunk 4]
```diff
@@ -24,6 +23,33 @@
     return fields;
   }
 
+  @org.testng.annotations.DataProvider(name = "fieldsOnBoundaries")
+  public static Object[] fieldsOnBoundaries() {
+
+    return  new Object[][]{ {0, 2}, { 0, 5}, {0, 7},
+              {9, 1}, { 9, 3}, {9, 6},
+              {2, 0}, { 4, 0}, {6, 0},
+              {1, 9}, { 3, 9}, {8, 9} };
```
Comment (Ozymadias) [inline client/src/test/java/model/DataProvider.java]: formatting - spaces before some numbers - maybe you want to change it

File: client/src/test/java/model/PositionTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    Position positionOne = new Position(index);
+    Position positionTwo = new Position(index);
+
+      //when
+      //then
+    assertTrue(positionOne.equals(positionTwo));
+  }
+
+  @Test(dataProvider = "positions")
+  public void positionsThatHasDifferentColAndRowAreEquals(int index) {
```
Comment (Ozymadias) [inline client/src/test/java/model/PositionTest.java]: does really points of your have different row AND column?

File: client/src/test/java/model/PossiblePositionsTest.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    sea = new Sea();
+    IntStream.range(0, 99).forEach(i -> sea.addSeaField(new DummyClickableField(i)));
+    possiblePositions = new PossiblePositions(sea);
+
+  }
+
+  @Test
+  public void shouldMakeClickableAllFieldsInProvidedList() {
+    List<ClickableField> fieldsToMakeClickable = new ArrayList<>();
+    ClickableField mockedFirstField = mock(ClickableField.class); //TODO: change mock to real object?
```
Comment (Ozymadias) [inline client/src/test/java/model/PossiblePositionsTest.java]: todo in production code?

File: client/src/test/java/model/PossiblePositionsTest.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+  @Test(dataProvider = "masts")
+  public void shouldReturnPositionsAvailableForMastPlacement(Field mast, Integer[] expectedPositions) {
+    List<ClickableField> actualAvailableFields = possiblePositions.findPositions(mast);
+    List<Integer> actualAvailablePositions = actualAvailableFields
+        .stream()
+        .map(Field::positionAsInteger)
+        .collect(Collectors.toList());
+
+    assertThat(actualAvailablePositions).containsExactlyInAnyOrder(expectedPositions);
```
Comment (Ozymadias) [inline client/src/test/java/model/PossiblePositionsTest.java]: nicely tested list :+1:

File: client/src/test/java/model/ShipCreatorTest.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+  @Test
+  public void whenShipIsNotFullyBuildMastIsAddedToShip() {
+
+    //given
+    int shipLength = 2;
+    Ship ship = new Ship(shipLength);
+    Field mast = mock(Field.class);
+
+    ShipCreator shipCreator = new ShipCreator(ship);
```
Comment (Ozymadias) [inline client/src/test/java/model/ShipCreatorTest.java]: you can extract to setUp method shipCreator, ship and mast - used in both test

File: client/src/main/java/model/Ship.java (hunk 1 above)
Comment (patiem) [reply client/src/main/java/model/Ship.java]: ok

File: client/src/test/java/model/DataProvider.java (hunk 2 above)
Comment (patiem) [reply client/src/test/java/model/DataProvider.java]: This is good point, I think we should ask Artem about that

File: client/src/test/java/model/DummyClickableField.java (hunk 3 above)
Comment (patiem) [reply client/src/test/java/model/DummyClickableField.java]: This is just dummy implementation for testing, my own mock :), we dont need body for this method, becouse we dont test ClickableField, but we use this implementation for testing other classes.

File: client/src/test/java/model/DataProvider.java (hunk 4 above)
Comment (patiem) [reply client/src/test/java/model/DataProvider.java]: ok

File: client/src/test/java/model/PositionTest.java (hunk 5 above)
Comment (patiem) [reply client/src/test/java/model/PositionTest.java]: I changed name of method, it was misleading.

File: client/src/test/java/model/PossiblePositionsTest.java (hunk 7 above)
Comment (patiem) [reply client/src/test/java/model/PossiblePositionsTest.java]: Kudos for Emilia :)

File: client/src/test/java/model/ShipCreatorTest.java (hunk 8 above)
Comment (patiem) [reply client/src/test/java/model/ShipCreatorTest.java]: I cant do that, because shipCreator gets different ships in constructor

**Pull request merged**